### DISCUSSION TITLE
ConfigMap and Ingress not getting deleted

### DISCUSSION THREAD

**Discussion by p4lik4ri** [k3s-io/k3s#7794] [Q&A]
**Environmental Info:**
K3s Version:  v1.26.5+k3s1

Node(s) CPU architecture, OS, and Version: 
5.19.0-43-generic ~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Mon May 22 13:39:36 UTC 2 x86_64 x86_64 x86_64 GNU/Linux

Cluster Configuration:
1 node and 0 agents

**Describe the bug:**

`kubectl delete all --all`  is not deleting all resources like ConfigMap and ingress resources. 

**Steps To Reproduce:**

- Installed K3s: ` curl -sfL https://get.k3s.io | sh -`

- Creating 4 manifests for nginx: nginx-deploy.yaml, nginx-service.yaml, nginx-config.yaml, nginx-ingress.yaml

- `kubectl apply -f ` each yaml file.
Terminal:
   _configmap/nginx-config created
   deployment.apps/nginx created
   ingress.networking.k8s.io/nginx-ingress created
   service/nginx-service created_

**Expected behavior:**

`kubectl delete all --all` would delete all of the above yaml file created. Deployment, service including ingress and configmap.

**Actual behavior:**

Delete all resources except configmap and ingress. 

`- kubectl delete all --all`

Terminal: 
              _pod "nginx-6c5496df94-f2zjr" deleted
                service "nginx-service" deleted
                service "kubernetes" deleted
                deployment.apps "nginx" deleted_

- `kubectl get cm`
Terminal:
                   kube-root-ca.crt   1      89m
                   nginx-config       1      9m10s_

- `kubectl get ingress`
Terminal:
                 nginx-ingress   <none>   *       192.168.xxx.xxx   80      15m

**Additional context / logs:**
 
According to official documentations ingress and configmap are resources and they are created under the same namespace as the deployments,services etc... (in my case default). So why they are not deleted together with the other resources? I am new in github and k8s so go easy on me :) 
Thanks in advanced !

**Comment by brandond** [comment]
If they are deleted but still show up in a list, there is likely a finalizer blocking their deletion. This usually indicates that there is some other resource controller that still depends on them, and has some clean up to do before they are safe to delete.

You can `kubectl get -o yaml` the resources in question and check the finalizers field to see what's holding on to references to them.

**Reply by p4lik4ri** [reply]
hello @brandond thanks for the quick response! For example i did `kubectl get ingress nginx-ingress -o yaml` and the outcome was the following:
```
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"networking.k8s.io/v1","kind":"Ingress","metadata":{"annotations":{"kubernetes.io/ingress.class":"traefik"},"name":"nginx-ingress","namespace":"default"},"spec":{"rules":[{"http":{"paths":[{"backend":{"service":{"name":"nginx-service","port":{"number":80}}},"path":"/","pathType":"Prefix"}]}}]}}
    kubernetes.io/ingress.class: traefik
  creationTimestamp: "2023-06-16T15:52:33Z"
  generation: 1
  name: nginx-ingress
  namespace: default
  resourceVersion: "2734"
  uid: 77ed4f4a-2af9-44a8-b421-ce52d95b7c3d
spec:
  rules:
  - http:
      paths:
      - backend:
          service:
            name: nginx-service
            port:
              number: 80
        path: /
        pathType: Prefix
status:
  loadBalancer:
    ingress:
    - ip: 192.168.xxx.xxx
```
it doesnt have any field called finalizer.

**Reply by brandond** [reply]
I'm not sure why it's not finding those resources to delete, but the `kubectl delete` behavior is not something we customize in k3s. I would take a look at the upstream docs and see if there is perhaps some nuance to that command that is being missed.
